From: Andrea Bolognani <abologna@redhat.com>
Date: Wed, 3 Nov 2021 10:32:14 +0100
Subject: meson: Stop looking up ZFS programs at build time

At this point, we're no longer using the availability of the
ZFS programs at build time to decide whether to enable ZFS
support, so the only purpose of these find_program() calls is
to record their absolute paths.

However, the virCommand facilities that we're ultimately using
to run them are already capable of performing this lookup at
runtime, and in fact that's exactly what we already do in the
case of, for example, vstorage.

Drop the build time lookups and always perform them at runtime.

Signed-off-by: Andrea Bolognani <abologna@redhat.com>
Reviewed-by: Tim Wiederhake <twiederh@redhat.com>
(cherry picked from commit 506c3a39d6e645c8414c278ceaba97935f90cb95)
---
 meson.build                       | 16 ----------------
 src/storage/storage_backend_zfs.c |  3 +++
 2 files changed, 3 insertions(+), 16 deletions(-)

diff --git a/meson.build b/meson.build
index 2dce8a1..5d299f2 100644
--- a/meson.build
+++ b/meson.build
@@ -1947,24 +1947,8 @@ if conf.has('WITH_LIBVIRTD')
   endif
 
   if not get_option('storage_zfs').disabled()
-    foreach name : [ 'zfs', 'zpool' ]
-      set_variable(
-        '@0@_prog'.format(name),
-        find_program(name, required: false, dirs: libvirt_sbin_path)
-      )
-    endforeach
-
     use_storage = true
     conf.set('WITH_STORAGE_ZFS', 1)
-    foreach name : [ 'zfs', 'zpool' ]
-      prog_var = get_variable('@0@_prog'.format(name))
-      if prog_var.found()
-        prog_path = prog_var.path()
-      else
-        prog_path = name
-      endif
-      conf.set_quoted(name.to_upper(), prog_path)
-    endforeach
   endif
 endif
 
diff --git a/src/storage/storage_backend_zfs.c b/src/storage/storage_backend_zfs.c
index 6fabead..2a5d743 100644
--- a/src/storage/storage_backend_zfs.c
+++ b/src/storage/storage_backend_zfs.c
@@ -33,6 +33,9 @@
 
 VIR_LOG_INIT("storage.storage_backend_zfs");
 
+#define ZFS "zfs"
+#define ZPOOL "zpool"
+
 /*
  * Some common flags of zfs and zpool commands we use:
  * -H -- don't print headers and separate fields by tab
